Source/SwiftLintFramework/Rules/UnhandledThrow.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  UnhandledThrow.swift
+//  SwiftLint
+//
+//  Created by Arthur Ariel Sabintsev on 2/16/17.
+//  Copyright © 2017 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct UnhandledThrowRule: Rule, ConfigurationProviderRule {
+
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "unhandled_throw",
+        name: "Unhandled Throw",
+        description: "When a throwing function does not explicitly handle a throw, the `throws` keyword can be removed.",
+        nonTriggeringExamples: [
+            "func f() throws {\n throw anError \n}\n",
+            "func f() throws { throw anError }\n",
+            "func f() throws -> Any {\n throw anError \n}\n",
+            "func f() throws -> Any { throw anError }\n",
+            "func f() throws {\n try anExpression \n}\n",
+            "func f() throws { try anExpression }\n",
+            "func f() throws -> Any {\n try anExpression \n}\n",
+            "func f() throws -> Any { try anExpression }\n",
+            "func f() throws {\n try? anExpression \n}\n",
+            "func f() throws { try? anExpression }\n",
+            "func f() throws -> Any {\n try? anExpression \n}\n",
+            "func f() throws -> Any { try? anExpression }\n",
+            "func f() throws {\n try! anExpression \n}\n",
+            "func f() throws { try! anExpression }\n",
+            "func f() throws -> Any {\n try! anExpression \n}\n",
+            "func f() throws -> Any { try! anExpression }\n"
+        ],
+        triggeringExamples: [
+            "func f() throws {}\n",
+            "func f() throws { }\n",
+            "func f() throws {\n}\n",
+            "func f() throws {\n \n}\n",
+            "func f() throws -> Any {}\n",
+            "func f() throws -> Any { }\n",
+            "func f() throws -> Any {\n}\n",
+            "func f() throws -> Any {\n \n}\n",
+            "func f() throws { var x = 0 }\n",
+            "func f() throws {\n var x = 0 \n}\n",
+            "func f() throws -> Any { var x = 0 }\n",
+            "func f() throws -> Any {\n var x = 0 \n}\n"
```

Comment:
What about a case like the one below? I would see this as a triggering example as the function `foo` will never throw.

```swift
func foo() throws {
  do {
     try bar()
  } catch {}
}
```